templates/cluster-template-internal.yaml

```diff
@@ -75,37 +73,36 @@ spec:
             path: /readyz
         instance:
           cpu:
-            sockets: ${VIRTINK_MACHINE_CPU_COUNT}
-            coresPerSocket: 1
+            sockets: 1
+            coresPerSocket: ${VIRTINK_CONTROL_PLANE_MACHINE_CPU_CORES:=2}
           memory:
-            size: ${VIRTINK_MACHINE_MEMORY_SIZE}
+            size: ${VIRTINK_CONTROL_PLANE_MACHINE_MEMORY_SIZE:=4Gi}
           kernel:
-            image: ${VIRTINK_MACHINE_KERNEL_IMAGE}
-            cmdline: console=ttyS0 root=/dev/vda rw
+            image: "${VIRTINK_CONTROL_PLANE_MACHINE_KERNEL_IMAGE:=smartxworks/capch-kernel-5.15.12}"
+            cmdline: "console=ttyS0 root=/dev/vda rw"
           disks:
             - name: rootfs
           interfaces:
             - name: pod
         volumes:
           - name: rootfs
             containerRootfs:
-              image: ${VIRTINK_MACHINE_ROOTFS_IMAGE}
-              size: ${VIRTINK_MACHINE_ROOTFS_SIZE}
+              image: "${VIRTINK_CONTROL_PLANE_MACHINE_ROOTFS_IMAGE:=smartxworks/capch-rootfs-1.24.0}"
+              size: ${VIRTINK_CONTROL_PLANE_MACHINE_ROOTFS_SIZE:=4Gi}
         networks:
           - name: pod
             pod: {}
 ---
 apiVersion: cluster.x-k8s.io/v1beta1
 kind: MachineHealthCheck
 metadata:
-  name: ${CLUSTER_NAME}-control-plane-unhealthy-5m
+  name: "${CLUSTER_NAME}-cp-unhealthy-5m"
 spec:
-  clusterName: ${CLUSTER_NAME}
-  maxUnhealthy: 50%
+  clusterName: "${CLUSTER_NAME}"
+  maxUnhealthy: 49%
```

Comment:
In fact, maxUnhealthy percentage should be set "49.999...%". When the percentage is not a whole number, the allowed number is rounded down, and the control plane replication must be an odd number, so 50% is ok.

Reply:
We set this percentage to avoid control plane remediation because we're sure the control plane cluster (actually the etcd cluster) can't be restored. When setting up control plane cluster the number of nodes in progress could be an even, 50% percentage will allow the remediation and let MHC controller try to replace the failed node. The remediation won't succeed, but it doesn't make any mistakes ! On the other hand, maybe the node restart in progress of setting up control plane is not what we focus on.

The percentage 49% will make mistakes in a large control plane cluster. For a 201 nodes control plane cluster, the max allowed unhealthy node will be 100 instead of 98.